apps/election-manager/src/lib/votecounting.ts

```diff
@@ -369,17 +369,22 @@ export const getContestTallyMeta = ({
       }
       return ((vote as unknown) as YesNoVote).length > 1
     })
-    const undervotes = contestVotes.filter((vote) => {
-      if (contest.type === 'candidate') {
-        return ((vote as unknown) as CandidateVote).length < contest.seats
-      }
-      return ((vote as unknown) as YesNoVote).length === 0
-    })
-
+    const numberOfUndervotes = contestVotes
+      .map((vote) => {
+        if (contest.type === 'candidate') {
+          const numVotesMarked = ((vote as unknown) as CandidateVote).length
+          if (numVotesMarked < contest.seats) {
+            return contest.seats - numVotesMarked
+          }
+          return 0
+        }
+        return ((vote as unknown) as YesNoVote).length === 0 ? 1 : 0
+      })
+      .reduce((a, b) => a + b, 0)
```

Comment:
If you do the cast to `Vote` earlier you can avoid having more than one of the `as unknown`. Also, you can just reduce without mapping first if you think it's clear enough:

```ts
    const contestVotes = contestCVRs.map(
      (cvr) => (cvr[contest.id] as unknown) as Vote
    )
    const overvotes = contestVotes.filter((vote) => {
      if (contest.type === 'candidate') {
        return vote.length > contest.seats
      }
      return vote.length > 1
    })
    const numberOfUndervotes = contestVotes.reduce((undervotes, vote) => {
      if (contest.type === 'candidate') {
        const numVotesMarked = vote.length
        if (numVotesMarked < contest.seats) {
          return undervotes + contest.seats - numVotesMarked
        }
        return undervotes
      }
      return vote.length === 0 ? 1 : 0
    }, 0)
```